emop-maven-plugin/src/main/java/edu/cornell/AffectedSpecsMojo.java

```diff
@@ -246,53 +252,68 @@ private static Path classFileToSource(Path classFile, Path classesDir, Path sour
 
     private void makeSourcesFile(String sourceList, Set<String> newClasses) throws MojoExecutionException {
         Set<String> classes = new HashSet<>();
-        Path testSourceDir = getTestSourceDirectory().toPath().toAbsolutePath();
-        Path sourceDir = Paths.get(getTestSourceDirectory().getAbsolutePath().replace(
-                "src" + File.separator + "test", "src" + File.separator + "main"));
+        List<String> compileSourceRoots = mavenProject.getCompileSourceRoots();
+        List<String> testCompileSourceRoots = mavenProject.getTestCompileSourceRoots();
+        List<Path> sourceDirs = Stream.concat(compileSourceRoots.stream(), testCompileSourceRoots.stream())
+                .map(path -> Paths.get(path).toAbsolutePath())
+                .collect(Collectors.toList());
 
+        classes:
```

Comment:
Didn't know about this feature until today! Thanks for teaching me something new :)